core/drivers/crypto/crypto_api/acipher/ecc.c

```diff
@@ -340,10 +341,139 @@ static TEE_Result ecc_shared_secret(struct ecc_keypair *private_key,
 	return ret;
 }
 
+static TEE_Result ecc_encrypt(struct ecc_public_key *key,
+			  const uint8_t *src, size_t src_len,
+			  uint8_t *dst, size_t *dst_len)
+{
+	TEE_Result ret = TEE_ERROR_BAD_PARAMETERS;
+	struct drvcrypt_encrypt_data edata = { };
+	struct drvcrypt_ecc *ecc = NULL;
+	size_t size_bytes = 0;
+
+	if (!key || !src || !dst) {
+		CRYPTO_TRACE("Input parameters reference error!\n");
+		return ret;
+	}
+
+	key->curve = TEE_ECC_CURVE_SM2;
+	size_bytes = get_ecc_key_size_bytes(key->curve);
+	if (!size_bytes) {
+		CRYPTO_TRACE("Invalid size_bytes!\n");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	/* Verify the ciphertext length function of the key size */
+	if (*dst_len < 2 * size_bytes + src_len + TEE_SM3_HASH_SIZE) {
+		CRYPTO_TRACE("Length (%zu) too short expected %zu bytes",
+			 *dst_len, 2 * size_bytes + src_len + TEE_SM3_HASH_SIZE);
+		*dst_len = 2 * size_bytes + src_len + TEE_SM3_HASH_SIZE;
+		CRYPTO_TRACE("Invalid src_len or dst_len!\n");
+		return TEE_ERROR_SHORT_BUFFER;
+	}
+
+	/* Uncompressed form indicator */
+	dst[0] = 0x04;
+
+	ecc = drvcrypt_get_ops(CRYPTO_ECC);
+	if (ecc && ecc->encrypt) {
+		/*
+		 * Prepare the encrypt structure data
+		 */
+		edata.key = key;
+		edata.size_sec = size_bytes;
+		edata.plaintext.data = (uint8_t *)src;
+		edata.plaintext.length = src_len;
+		edata.ciphertext.data = (uint8_t *)dst + 1;
+		edata.ciphertext.length = *dst_len;
+
+		ret = ecc->encrypt(&edata);
+
+		/* Set the ciphertext length */
+		*dst_len = edata.ciphertext.length + 1;
+	} else {
+		ret = TEE_ERROR_NOT_IMPLEMENTED;
+	}
+
+	return ret;
+}
+
+static TEE_Result ecc_decrypt(struct ecc_keypair *key,
+			  const uint8_t *src, size_t src_len,
+			  uint8_t *dst, size_t *dst_len)
+{
+	TEE_Result ret = TEE_ERROR_BAD_PARAMETERS;
+	struct drvcrypt_encrypt_data edata = { };
+	struct drvcrypt_ecc *ecc = NULL;
+	uint8_t *ciphertext = NULL;
+	size_t ciphertext_len = 0;
+	size_t size_bytes = 0;
+	size_t C1_len = 0;
+	size_t C2_len = 0;
+	/* Point Compression */
+	uint8_t PC = 0;
+
+	if (!key || !src || !dst) {
+		CRYPTO_TRACE("Input parameters reference error!\n");
+		return ret;
+	}
+
+	key->curve = TEE_ECC_CURVE_SM2;
```

Comment:
Why set this curve by default in the crypto API? The API should better check if the given `TEE_ECC_CURVE_SM2` is supported or not.